source/extensions/common/matcher/matcher.cc

```diff
@@ -64,6 +105,53 @@ void buildMatcher(const envoy::config::common::matcher::v3::MatchPredicate& matc
   matchers[new_matcher->index()] = std::move(new_matcher);
 }
 
+// TODO(shivanshu21): Once HeaderMatcher is deprecated in favor of StringMatcher,
```

Comment:
Again, this comment is very confusing, since we are not deprecated HeaderMatcher in some some future PR in favor of StringMatcher, _this_ is the PR that does it. What we're going to do is move away from using the matcher in `envoy.config` to the new one in `envoy.type`.